### ISSUE TITLE
Function constructor is *not* a form of eval when used without arguments.

### ISSUE THREAD

**Issue by ponychicken** [jshint/jshint#1619]
As far as I know

```
var a = new Function();
```

is the same as 

```
var a = function() {};
```

**Comment by lauriro** [comment]
`new Function()` is not same as `function() {}`

Functions defined by function expressions are parsed only once,
while those defined by the Function constructor must be parsed each and every time the constructor is called. Calling a heavy compiler in the middle of run-time is bad idea.

Please do not use `new Function()` in this way, it would be same as

```
var undef = eval()
, ten = eval("10")
// I am not going to make this uglier by writing a loop example
```

**Comment by sunilkumarc** [comment]
I agree with @lauriro. new Function() is close to eval(). So it can change lexical scope of variables in the runtime. As a result, compiler will not optimize the code when converting to machine level code.
